connectors/class-connector-woocommerce.php

```diff
@@ -437,8 +437,8 @@ public function callback_deleted_post( $post_id ) {
 			return;
 		}
 
-		$order           = new \WC_Order( $post->ID );
-		$order_title     = esc_html__( 'Order number', 'stream' ) . ' ' . esc_html( $order->get_order_number() );
+		$order_id        = \WC_Order_Factory::get_order_id( $post->ID );
```

Comment:
<!-- potential concern, ask for clarification or investigation -->
**❓question**
I'm not sure on this -- `WC_Order_Factory::get_order_id()` will always return the post ID: 

```	
        /**
	 * Get the order ID depending on what was passed.
	 *
	 * @since 3.0.0
	 * @param  mixed $order Order data to convert to an ID.
	 * @return int|bool false on failure
	 */
	public static function get_order_id( $order ) {
		if ( false === $order ) {
			return self::get_global_order_id();
		} elseif ( is_numeric( $order ) ) {
			return $order;
		} elseif ( $order instanceof WC_Abstract_Order ) {
			return $order->get_id();
		} elseif ( ! empty( $order->ID ) ) {
			return $order->ID;
		} else {
			return false;
		}
	}
```

Is that the order ID now? 

Although -- this is on the `deleted_post` hook so it should be ok?

Reply:
@tharsheblows good question, however I agree that we should be OK since this action is triggered on post deletion. 

While I was working on a solution, I came across this comment on the [abstract-wc-order.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/abstracts/abstract-wc-order.php#L111-L118), and that's why I used it.

```
* Get the order if ID is passed, otherwise the order is new and empty.
* This class should NOT be instantiated, but the wc_get_order function or new WC_Order_Factory
* should be used. It is possible, but the aforementioned are preferred and are the only
* methods that will be maintained going forward.
 ```

Reply:
@marcinkrzeminski Ah ha, thanks!